src/compile/output.rs

```diff
@@ -0,0 +1,58 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ContractOutput {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    abi: Option<Vec<ContractFunctionOutput>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    bin: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    metadata: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    devdoc: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    userdoc: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none", rename = "kebab-case")]
+    storage_layout: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    ast: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    asm: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none", rename = "kebab-case")]
+    bin_runtime: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    hashes: Option<HashMap<String, String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    factory_deps: Option<HashMap<String, String>>,
+}
```

Comment:
Why is every field optional? Do you have the docs for this schema?

Reply:
At this moment, we only support one output type obtained with the `--combined-json` flag. This flag supports many arguments and combines the response depending on which arguments you use. If you use only one argument like `abi`, you will have only the `abi` field in the response. I don't think there's good documentation about this, but you can test it using the CLI with the `--combined-json` flag and passing arguments like `abi,bin,hashes`.